python_modules/dagster/dagster/_core/definitions/assets.py

```diff
@@ -296,7 +332,9 @@ def __init__(
         self._specs_by_key = {spec.key: spec for spec in normalized_specs}
 
         self._partition_mappings = get_partition_mappings_from_deps(
-            {}, [dep for spec in normalized_specs for dep in spec.deps], node_def.name
+            {},
+            [dep for spec in normalized_specs for dep in spec.deps],
+            node_def.name if node_def else "external assets",
```

Comment:
@sryza as a follow up can you change these to kwargs and rename the third parameter to something that makes it clear that this is generating a name for display purpose only, and that we aren't hardcoding the asset name to "external assets"